src/jit/emitarm64.cpp

```diff
@@ -7379,7 +7379,9 @@ void emitter::emitIns_Call(EmitCallType          callType,
 
 /*static*/ emitter::code_t emitter::insEncodeDatasizeLS(emitter::code_t code, emitAttr size)
 {
-    if (code & 0x00800000) // Is this a sign-extending opcode? (i.e. ldrsw, ldrsh, ldrsb)
+    bool exclusive = ((code & 0x35000000) == 0);
+
+    if ((code & 0x00800000) && !exclusive) // Is this a sign-extending opcode? (i.e. ldrsw, ldrsh, ldrsb)
```

Comment:
The original code was not fully decoding whether this was a sign extending load.

Acquire/Release are in the exclusive group which does not support sign extending forms.  So this is needed to correctly generate some of the store release forms..

The decode here is still incomplete, but it is sufficient for our current set of generated instructions.  I might expect similar issues when we try to support ARM 8.1 atomics...